client/hotrod-client/src/main/java/org/infinispan/client/hotrod/impl/RemoteCacheValuesCollection.java

```diff
@@ -0,0 +1,82 @@
+package org.infinispan.client.hotrod.impl;
+
+import java.util.AbstractCollection;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Spliterator;
+import java.util.stream.Stream;
+
+import org.infinispan.commons.util.CloseableIterator;
+import org.infinispan.commons.util.CloseableIteratorCollection;
+import org.infinispan.commons.util.CloseableSpliterator;
+import org.infinispan.commons.util.Closeables;
+import org.infinispan.commons.util.IntSet;
+import org.infinispan.commons.util.IteratorMapper;
+import org.infinispan.commons.util.RemovableCloseableIterator;
+
+class RemoteCacheValuesCollection<V> extends AbstractCollection<V> implements CloseableIteratorCollection<V> {
+   private InternalRemoteCache<?, V> remoteCache;
+   private final IntSet segments;
+
+   RemoteCacheValuesCollection(InternalRemoteCache<?, V> remoteCache, IntSet segments) {
+      this.remoteCache = remoteCache;
+      this.segments = segments;
+   }
+
+   @Override
+   public CloseableIterator<V> iterator() {
+      CloseableIterator<Map.Entry<?, V>> entryIterator = ((InternalRemoteCache) remoteCache).entryIterator(segments);
```

Comment:
you can avoid the cast if you add `K` as a type: `RemoteCacheValuesCollection<K, V>`

Reply:
I can't add type K as it then doesn't override the method.

Reply:
If you change the class signature to `RemoteCacheValuesCollection<K, V>`, and the `remoteCache` filter, it works.

Reply:
Sorry, yes I was only thinking of the method. I can add it to the class definition sure.